src/frontend/src/app/core/state/proposal/proposal.service.ts

```diff
@@ -37,66 +41,85 @@ export class ProposalService {
 
   constructor(private readonly actorService: BackendActorService) {}
 
-  public async loadOpenProposalList(): Promise<void> {
-    let getResponse: ListProposalsResponse;
+  public async loadProposalList(state?: ReviewPeriodState): Promise<void> {
+    if (state) {
+      if ('in_progress' in state) {
+        this.loadOpenProposals(state);
+      } else if ('completed' in state) {
+        this.loadClosedProposals(state);
+      }
+    } else if (state === undefined) {
+      this.loadAllProposals();
+    }
+  }
 
-    if (!this.isCached()) {
-      getResponse = await this.actorService.list_proposals({
-        state: [{ in_progress: null }],
-      });
-    } else return;
+  public async setCurrentProposalId(proposalId: bigint): Promise<void> {
+    this.currentProposalIdSubject.next(proposalId);
+  }
 
-    if (isOk(getResponse)) {
-      this.openProposalListSubject.next(
-        mapProposalListResponse(getResponse.ok.proposals),
-      );
-      this.lastLoadTime = Date.now();
+  private async loadOpenProposals(state: ReviewPeriodState): Promise<void> {
+    if (this.isCached(this.openProposalListLastLoaded)) {
+      this.currentProposalListSubject.next(this.openProposalList);
       return;
+    } else {
+      const getResponse = await this.actorService.list_proposals({
+        state: optional(state),
```

Comment:
State should always be `'in_progress'`, so does it make sense to pass it in as a parameter?